pkg/integrations/windows_exporter/config.go

```diff
@@ -0,0 +1,128 @@
+package windows_exporter //nolint:golint
+
+import (
+	"reflect"
+
+	"github.com/go-kit/kit/log"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/config"
+)
+
+// Config controls the node_exporter integration.
+type Config struct {
+	Common config.Common `yaml:",inline"`
+
+	EnabledCollectors string `yaml:"enabled_collectors"`
+
+	Exchange    *ExchangeConfig    `yaml:"exchange"`
+	IIS         *IISConfig         `yaml:"iis"`
+	TextFile    *TextFileConfig    `yaml:"text_file"`
+	SMTP        *SMTPConfig        `yaml:"smtp"`
+	Service     *ServiceConfig     `yaml:"service"`
+	Process     *ProcessConfig     `yaml:"process"`
+	Network     *NetworkConfig     `yaml:"network"`
+	MSSQL       *MSSQLConfig       `yaml:"mssql"`
+	MSMQ        *MSMQConfig        `yaml:"msmq"`
+	LogicalDisk *LogicalDiskConfig `yaml:"logical_disk"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config.
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+func (c *Config) Name() string {
+	return "windows_exporter"
+}
+
+func (c *Config) CommonConfig() config.Common {
+	return c.Common
+}
+
+func (c *Config) NewIntegration(l log.Logger) (integrations.Integration, error) {
+	return New(l, c)
+}
+
+func (c *Config) ConvertToMap() map[string]string {
+	configMap := make(map[string]string)
+	mapToConfig(c, configMap)
+	return configMap
+}
+
+// This is to create a map from the config using the exporter struct tag as the key. This is to be ONLY used on pointer configs
+// and pointer strings. This is because we need to know if a value was set so that we don't overwrite defaults.
+func mapToConfig(config interface{}, cm map[string]string) {
```

Comment:
This feels like technical debt and would love any feedback. Also going to check it out to see if I can simplify/harden it.